pkg/lib/packagekit.es6

```diff
@@ -264,3 +264,83 @@ export function watchRedHatSubscription(callback) {
             }
         );
     }
+
+/* Support for installing optional packages.
+ *
+ * First call check_optional_packages to determine whether something
+ * needs to be installed, then call install_optional_packages to
+ * actually install them.
+ *
+ * check_optional_packages resolves to either null when no packages
+ * need to be installed, or an object that can be passed to
+ * install_optional_packages.
+ *
+ * In the future, that object might contain more information about
+ * how many packages need to be installed, and how much space that
+ * will take, but, PackageKit...
+ */
+
+export function check_optional_packages(names, progress_cb) {
```

Comment:
"optional" is IMHO a bit confusing - we are using this for use cases where said packages are actually required. How about `check_missing_packages()`?

Reply:
Yes, very good!  I didn't like "optional" either.